Feat: add sindy

Faezehabibi opened the pull request.

Faezehabibi committed:
Create std_sindy.py

Faezehabibi committed:
feat: add train_sindy.py

Faezehabibi committed:
Update std_sindy.py

Faezehabibi committed:
Update train_sindy.py

Faezehabibi committed:
Update train_sindy.py

Faezehabibi committed:
Update train_sindy.py

refactor PolynomialLibrary path

Faezehabibi committed:
Update train_sindy.py

Faezehabibi committed:
Update train_sindy.py

Faezehabibi commented:
now updated

ago109 commented on exhibits/sindy/train_sindy.py:
Change Line 5 `from sindy import Std_SINDy` to `from std_sindy import Std_SINDy`

ago109 committed:
Update std_sindy.py

Ported over and revised code from ngc-museum-faeze

ago109 committed:
Rename std_sindy.py to sindy.py

ago109 committed:
Update train_sindy.py

Ported over train_sindy.py from ngc-museum-faeze and added in corrections/revisions for museum compliance

ago109 reviewed (commented):
Ported over most up-to-date files and revised them based on what was in `ngc-museum-faeze` repo
